src/panels/edit-panel/panelCtrl.ts

```diff
@@ -145,6 +156,7 @@ export class PlotlyPanelCtrl extends MetricsPanelCtrl {
   }
 
   onDataReceived(dataList) {
+    this.queryList = dataList;
```

Comment:
Non mi piace che ci sia il `queryList` come attributo della classe, se risolviamo del bottone sarebbe meglio toglierlo. Piuttosto si potrebbe fare la classe `update_queries()` parametrica che accetta in input il dataList e basta, a fine funzione metti un `this.refresh();` (oltre che al clone che ti ho scritto sopra) e vedi se ti risolve il problema.

Reply:
Visto che la dataList è disponibile solo quando viene chiamato il metodo onDataReceived per usare il bottone è necessario salvarla da qualche parte. Devo provare meglio con influx ma pare che Grafana chiami il metodo molte volte, facendo piantare tutto

Reply:
Lo chiama ogni volta che c'è un refresh, all'avvio mi sembra due volte. Sì dataList dicevo di toglierlo se si riusciva togliere il pulsante dalla grafica